File: SDKs/netboxdemo.com/2.4/go/pkg/models/operations/ipamipaddressesread.go
package operations

import (
	"openapi/pkg/models/shared"
)

type IpamIPAddressesReadPathParams struct {
	ID int64 `pathParam:"style=simple,explode=false,name=id"`
}

type IpamIPAddressesReadRequest struct {
	PathParams IpamIPAddressesReadPathParams
}

type IpamIPAddressesReadResponse struct {
	ContentType string
	IPAddress   *shared.IPAddress
	StatusCode  int64
}

File: SDKs/netboxdemo.com/2.4/typescript/src/sdk/models/operations/ipamipaddressesread.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";


export class IpamIpAddressesReadPathParams extends SpeakeasyBase {
  @Metadata({ data: "pathParam, style=simple;explode=false;name=id" })
  id: number;
}


export class IpamIpAddressesReadRequest extends SpeakeasyBase {
  @Metadata()
  pathParams: IpamIpAddressesReadPathParams;
}


export class IpamIpAddressesReadResponse extends SpeakeasyBase {
  @Metadata()
  contentType: string;

  @Metadata()
  ipAddress?: shared.IpAddress;

  @Metadata()
  statusCode: number;
}
